src/PrestaShopBundle/Controller/Admin/Sell/Catalog/AttachmentController.php

```diff
@@ -277,24 +286,25 @@ public function searchAction(string $searchPhrase): JsonResponse
 
         $result = [];
         foreach ($attachments as $attachment) {
-            $result[] = $this->presentAttachmentInfo($attachment);
+            $result[] = $this->presentAttachmentInfo($attachment, $languageContext);
         }
 
         return $this->json($result);
     }
 
     /**
      * @param AttachmentInformation $productAttachmentInfo
+     * @param LanguageContext $languageContext
      *
      * @return array<string, mixed>
      */
-    private function presentAttachmentInfo(AttachmentInformation $productAttachmentInfo): array
+    private function presentAttachmentInfo(AttachmentInformation $productAttachmentInfo, LanguageContext $languageContext): array
     {
         $localizedNames = $productAttachmentInfo->getLocalizedNames();
 
         return [
             'attachment_id' => $productAttachmentInfo->getAttachmentId(),
-            'name' => $localizedNames[$this->getContextLangId()] ?? reset($localizedNames),
+            'name' => $localizedNames[$languageContext->getId()] ?? reset($localizedNames),
```

Comment:
```suggestion
            'name' => $localizedNames[$this->getLanguageContext()->getId()] ?? reset($localizedNames),
```
You have helper methids now to access all the context services, so you don't even need to inject it in the action or to pass via the method, much easier to use (you may need to rebase your branch if it's not up to date it was merged recently)